### PR TITLE
Fix georges

### PR DESCRIPTION
**Pull request by snim2**
**Needs squashing**

This PR is a simplified version of the code in the `georges_more_detailed` branch. It contributes the following:

  * Fixes the core algorithm that implemented the method in Georges et al.
  * Fixes a bug in the tables (due to not always sorting the rows by machine name)
  * Adds a feature to write out summary macros of the data in the tables

Unlike the other branch, this one does not include the "more detailed" tables with per-benchmark data.

### Testing

Since the code in the other branch is buggy, the output of this script now needs to be tested against the data it collects. For the v0.8 data, these are the raw counts:

```python
{u'bencher5': {'flat': 357, 'total no steady state': 120, 'total warmup': 666, 'warmup': 666, 'total flat': 364, 'no steady state': 100, 'total slowdown': 230, 'slowdown': 229}, 

u'bencher7': {'flat': 289, 'total no steady state': 133, 'total warmup': 711, 'warmup': 711, 'total flat': 289, 'no steady state': 102, 'total slowdown': 247, 'slowdown': 247}, 

u'bencher6': {'flat': 303, 'total no steady state': 25, 'total warmup': 469, 'warmup': 465, 'total flat': 306, 'no steady state': 15, 'total slowdown': 100, 'slowdown': 98}}
``` 

This is the table: [new_pr.pdf](https://github.com/softdevteam/warmup_experiment/files/930521/new_pr.pdf)

And these are the summary macros:

```latex
%%%
%%% Number of benchmarks on that Georges et al. found to reach a state state.
%%%
\newcommand{\georgesbencherfiveflat}{357}
\newcommand{\georgesbencherfivewarmup}{666}
\newcommand{\georgesbencherfiveslowdown}{229}
\newcommand{\georgesbencherfivenosteadystate}{100}
\newcommand{\georgesbenchersevenflat}{289}
\newcommand{\georgesbenchersevenwarmup}{711}
\newcommand{\georgesbenchersevenslowdown}{247}
\newcommand{\georgesbenchersevennosteadystate}{102}
\newcommand{\georgesbenchersixflat}{303}
\newcommand{\georgesbenchersixwarmup}{465}
\newcommand{\georgesbenchersixslowdown}{98}
\newcommand{\georgesbenchersixnosteadystate}{15}
%%%
%%% Percentage of benchmarks on that Georges et al. found to reach a state state.
%%%
\newcommand{\georgesbencherfiveflatpercent}{98.1}
\newcommand{\georgesbencherfivewarmuppercent}{100.0}
\newcommand{\georgesbencherfiveslowdownpercent}{99.6}
\newcommand{\georgesbencherfivenosteadystatepercent}{83.3}
\newcommand{\georgesbenchersevenflatpercent}{100.0}
\newcommand{\georgesbenchersevenwarmuppercent}{100.0}
\newcommand{\georgesbenchersevenslowdownpercent}{100.0}
\newcommand{\georgesbenchersevennosteadystatepercent}{76.7}
\newcommand{\georgesbenchersixflatpercent}{99.0}
\newcommand{\georgesbenchersixwarmuppercent}{99.1}
\newcommand{\georgesbenchersixslowdownpercent}{98.0}
\newcommand{\georgesbenchersixnosteadystatepercent}{60.0}
%%%
%%% Percentage of benchmarks over all machines that Georges et al. found to reach a state state.
%%%
\newcommand{\georgesflatpercent}{99.0}
\newcommand{\georgeswarmuppercent}{99.8}
\newcommand{\georgesslowdownpercent}{99.5}
\newcommand{\georgesnosteadystatepercent}{78.1}
```

### CODE REVIEW

File: bin/table_georges_comparison [hunk 1]
```diff
@@ -102,14 +111,64 @@ def write_tables(counts, tex_filename, with_preamble=False):
...
             fp.write(end_document())
 
 
+def _reformat(word):
+    """Reformat word as a LaTeX macro name.
+    Removes spaces and underscores. Translates numbers to words. Lower cases.
+    """
+    for number in _NUMBERS:
+        word = word.replace(str(number), _NUMBERS[number])
+    word = word.replace(' ', '')
```
Comment (ltratt) [inline bin/table_georges_comparison]: This line and the one below may be clearer as `word = word.translate(None, " _")`?

Comment (snim2) [inline bin/table_georges_comparison]: There is a space in the first string...

Comment (ltratt) [reply bin/table_georges_comparison]: There's one in the string I put in too ;) It's `" _"` i.e. quote space underscore quote.

**Comment by ltratt** [comment]
Looks good to me. Do you want to squash then I can merge?

**Comment by snim2** [comment]
Just a moment, there's something wrong with that call to `translate()`.

**Comment by snim2** [comment]
OK, ready to squash?

**Comment by ltratt** [comment]
Yep! [And "sigh" on `translate` and unicode strings :( I wonder if that's fixed in Python 3?]

**Comment by snim2** [comment]
I wouldn't even pretend to understand unicode support in Python 3!

**Comment by snim2** [comment]
Splat.

**Review by ltratt — APPROVED** [review]

**Pull request merged**